// react-mdl/lib/Grid.js
'use strict';

exports.__esModule = true;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = (function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ('value' in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; })();

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError('Cannot call a class as a function'); } }

function _inherits(subClass, superClass) { if (typeof superClass !== 'function' && superClass !== null) { throw new TypeError('Super expression must either be null or a function, not ' + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var Grid = (function (_React$Component) {
    _inherits(Grid, _React$Component);

    function Grid() {
        _classCallCheck(this, Grid);

        _React$Component.apply(this, arguments);
    }

    Grid.prototype.render = function render() {
        var _props = this.props;
        var className = _props.className;

        var otherProps = _objectWithoutProperties(_props, ['className']);

        var classes = _classnames2['default']('mdl-grid', className);

        return _react2['default'].createElement(
            'div',
            _extends({ className: classes }, otherProps),
            this.props.children
        );
    };

    _createClass(Grid, null, [{
        key: 'propTypes',
        value: {
            className: _react.PropTypes.string
        },
        enumerable: true
    }]);

    return Grid;
})(_react2['default'].Component);

exports['default'] = Grid;

var Cell = (function (_React$Component2) {
    _inherits(Cell, _React$Component2);

    function Cell() {
        _classCallCheck(this, Cell);

        _React$Component2.apply(this, arguments);
    }

    Cell.prototype.render = function render() {
        var _classNames;

        var _props2 = this.props;
        var className = _props2.className;
        var col = _props2.col;
        var phone = _props2.phone;
        var tablet = _props2.tablet;

        var otherProps = _objectWithoutProperties(_props2, ['className', 'col', 'phone', 'tablet']);

        var classes = _classnames2['default']('mdl-cell', (_classNames = {}, _classNames['mdl-cell--' + col + '-col'] = true, _classNames['mdl-cell--' + phone + '-col-tablet'] = typeof phone !== 'undefined', _classNames['mdl-cell--' + tablet + '-col-tablet'] = typeof tablet !== 'undefined', _classNames), className);

        return _react2['default'].createElement(
            'div',
            _extends({ className: classes }, otherProps),
            this.props.children
        );
    };

    _createClass(Cell, null, [{
        key: 'propTypes',
        value: {
            className: _react.PropTypes.string,
            col: _react.PropTypes.number.isRequired,
            phone: _react.PropTypes.number,
            tablet: _react.PropTypes.number
        },
        enumerable: true
    }]);

    return Cell;
})(_react2['default'].Component);

exports['default'] = Grid;
exports.Cell = Cell;
