// react-mdl/lib/tabs/Tabs.js
'use strict';

exports.__esModule = true;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = (function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ('value' in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; })();

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError('Cannot call a class as a function'); } }

function _inherits(subClass, superClass) { if (typeof superClass !== 'function' && superClass !== null) { throw new TypeError('Super expression must either be null or a function, not ' + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _Tab = require('./Tab');

var _Tab2 = _interopRequireDefault(_Tab);

var Tabs = (function (_React$Component) {
    _inherits(Tabs, _React$Component);

    function Tabs() {
        var _this = this;

        _classCallCheck(this, Tabs);

        _React$Component.apply(this, arguments);

        this._handleClickTab = function (tabId) {
            _this.props.onChange(tabId);
        };
    }

    Tabs.prototype.render = function render() {
        var _this2 = this;

        var _props = this.props;
        var activeTab = _props.activeTab;
        var className = _props.className;
        var onChange = _props.onChange;

        var otherProps = _objectWithoutProperties(_props, ['activeTab', 'className', 'onChange']);

        var classes = _classnames2['default']('mdl-tabs is-upgraded', className);

        return _react2['default'].createElement(
            'div',
            _extends({ className: classes }, otherProps),
            _react2['default'].createElement(
                'div',
                { className: 'mdl-tabs__tab-bar' },
                _react2['default'].Children.map(this.props.children, function (child, index) {
                    return _react2['default'].cloneElement(child, {
                        tabId: index,
                        active: index === (activeTab || 0),
                        onTabClick: _this2._handleClickTab
                    });
                })
            )
        );
    };

    _createClass(Tabs, null, [{
        key: 'propTypes',
        value: {
            activeTab: _react.PropTypes.number,
            children: _react.PropTypes.arrayOf(function (props, propName, componentName) {
                var prop = props[propName];
                if (prop.type !== _Tab2['default']) {
                    return new Error('`' + componentName + '` only accepts `Tab` as children.');
                }
            }),
            className: _react.PropTypes.string,
            onChange: _react.PropTypes.func.isRequired
        },
        enumerable: true
    }]);

    return Tabs;
})(_react2['default'].Component);

exports['default'] = Tabs;
module.exports = exports['default'];
